src/hw/aarch64_virt/port.c

```diff
@@ -22,3 +22,10 @@ void print_register_context(const RegisterContext* ctx) {
          ctx->x22, ctx->x23, ctx->x24, ctx->x25, ctx->x26, ctx->x27, ctx->x28,
          ctx->x29, ctx->x30, ctx->pc, ctx->spsr_el1, ctx->fpsr);
 }
+
+static volatile uint32_t* const UART0 = (uint32_t*)UART_BASE;
+
+void platform_putchar(int c) {
+  *UART0 = (uint32_t)c;
+  // We do not modify buf here, serial port doesn't move
```

Comment:
The comment was misleading. Can you address the problems you encountered?

Reply:
This comment is used to address the difference between `SerialPrintOutput::write` and `BufferPrintOutput::write`.
But it actually becomes useless right here. So I will remove that.